File: leetcode/algorithms/math/angle-between-hands-of-a-clock.java
/*
    leetcode
    category: "Algorithms"
    problem title: "Angle Between Hands of a Clock"
*/

class Solution {
    public double angleClock(int hour, int minutes) {
        double hourAbsoluteAngle = (hour%12)*30+((double)minutes/60)*30;
        double minAbsoluteAngle = (minutes%60)*6;
        
        double absoluteDiff = Math.abs(hourAbsoluteAngle-minAbsoluteAngle);
        
        return absoluteDiff>180 ? 360-absoluteDiff : absoluteDiff;
        
    }
}

File: leetcode/algorithms/math/angle-between-hands-of-a-clock.js
/*
    leetcode
    category: "Algorithms"
    problem title: "Angle Between Hands of a Clock"
*/

/**
 * @param {number} hour
 * @param {number} minutes
 * @return {number}
 */
var angleClock = function(hour, minutes) {
    
    const angleBetweenHands = Math.abs((hour%12)*30 + (minutes/60)*30-minutes*6);
    
    const angleBetweenHands = Math.abs(hourAbsoluteAngle-minuteAbsoluteAngle);
    return angleBetweenHands>180 ? 360-angleBetweenHands : angleBetweenHands;
    
};